hrm-domain/hrm-core/jobs/search/publishToSearchIndex.ts

```diff
@@ -19,11 +19,41 @@ import { getSsmParameter } from '../../config/ssmCache';
 import { IndexMessage } from '@tech-matters/hrm-search-config';
 import { CaseService, Contact } from '@tech-matters/hrm-types';
 import { AccountSID, HrmAccountId } from '@tech-matters/types';
+import { publishSns, PublishSnsParams } from '@tech-matters/sns-client';
+import { SsmParameterNotFound } from '@tech-matters/ssm-cache';
+
+type DeleteNotificationPayload = {
+  accountSid: HrmAccountId;
+  operation: 'delete';
+  id: string;
+};
+
+type UpsertCaseNotificationPayload = {
+  accountSid: HrmAccountId;
+  operation: 'update' | 'create' | 'reindex';
+  case: CaseService;
+};
+
+type UpsertContactNotificationPayload = {
+  accountSid: HrmAccountId;
+  operation: 'update' | 'create' | 'reindex';
+  contact: Contact;
+};
+
+type NotificationPayload =
+  | DeleteNotificationPayload
+  | UpsertCaseNotificationPayload
+  | UpsertContactNotificationPayload;
 
 const PENDING_INDEX_QUEUE_SSM_PATH = `/${process.env.NODE_ENV}/${
   process.env.AWS_REGION ?? process.env.AWS_DEFAULT_REGION
 }/sqs/jobs/hrm-search-index/queue-url-consumer`;
 
+const getSnsSsmPath = dataType =>
+  `/${process.env.NODE_ENV}/${
+    process.env.AWS_REGION ?? process.env.AWS_DEFAULT_REGION
+  }/sns/hrm/${dataType}/update-notifications-sns-topic`;
```

Comment:
Wouldn't be easier to provide the SSM parameter as env var via `.tf`? Applies to the indexing queue as well, that I see is using the same format as this one :P
Maybe for a next PR?